lib/io/hyped_spi/hyped_spi_derive/src/lib.rs

```diff
@@ -13,7 +13,68 @@ pub fn hyped_spi_derive(input: TokenStream) -> TokenStream {
 fn impl_hyped_spi(ast: &syn::DeriveInput) -> TokenStream {
     let name = &ast.ident;
     let generics: &Generics = &ast.generics;
-    let (_impl_generics, _ty_generics, _) = generics.split_for_impl();
-    let gen = quote! {};
+    let (impl_generics, ty_generics, _) = generics.split_for_impl();
+    let gen = quote! {
+        impl #impl_generics HypedSpi for #name #ty_generics{
+            fn read(&mut self, words: &mut [u8]) -> Result<(), SpiError> {
+                let mut binding = Vec::<u8, 64>::new();
+                let new_words = binding.as_mut_slice();
+                match self.spi.blocking_read(new_words) {
+                    Ok(_) => {
+                        for (i, word) in new_words.iter().enumerate() {
+                            words[i] = *word;
```

Comment:
check if `new_words` will actually fit in `words`

or is there a 64 byte limit on spi reads?